src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
+
+  static addAliveAnimals(item) {
+    Animal.alive.push(item);
+  }
+
+  static removeAnimals() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    Animal.addAliveAnimals(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.addAliveAnimals(this);
+  }
+
+  bite(item) {
+    if (item instanceof Herbivore && !item.hidden) {
+      item.health -= 50;
+      Animal.removeAnimals();
```

Comment:
Consider moving the `Animal.removeAnimals` call outside of the `if` statement in the `bite` method. It should be called regardless of whether a bite occurred or not to ensure the `Animal.alive` array is always up-to-date.